### PR TITLE
Add connection profile persistence to MIAA dashboard

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Hooks up persisting the connection profile we get back from the connection dialog when the user opens the dashboard. We utilize the connection storage in core to reduce the amount of duplicated info we have to store - plus this way we get benefits like being able to have the connection added to the server view immediately if the user wants.

Some more tweaks I'll be following up with in later PRs:
1. Populate the server name in the connection dialog when it opens
2. Don't allow editing certain fields (like server name) in the connection dialog
3. Handling connection errors better

### CODE REVIEW

File: extensions/arc/src/localizedConstants.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const newSupportRequest = localize('arc.newSupportRequest', "New support
 export const diagnoseAndSolveProblems = localize('arc.diagnoseAndSolveProblems', "Diagnose and solve problems");
 export const supportAndTroubleshooting = localize('arc.supportAndTroubleshooting', "Support + troubleshooting");
 
-export const createNew = localize('arc.createNew', "Create New");
+export const newInstance = localize('arc.createNew', "New Instance");
```
Comment (Charles-Gagnon) [inline extensions/arc/src/localizedConstants.ts:32]: Guess some extra cleanup changes snuck in here - whoops. Oh well, it's pretty much just renaming the button to be more descriptive of what it's doing.

File: extensions/arc/src/ui/tree/controllerTreeNode.ts [hunk 2]
```diff
@@ -51,21 +53,44 @@ export class ControllerTreeNode extends TreeNode {
...
-				case ResourceType.sqlManagedInstances:
-					const miaaModel = new MiaaModel(this.model.info.url, this.model.auth!, registration.instanceNamespace, parseInstanceName(registration.instanceName));
-					return new MiaaTreeNode(miaaModel, this.model);
+
+			const resourceInfo: ResourceInfo = {
+				namespace: registration.instanceNamespace,
+				name: parseInstanceName(registration.instanceName)
+			};
+
+			let node = this._children.find(n => n.model?.info?.name === resourceInfo.name && n.model?.info?.namespace === resourceInfo.namespace);
```
Comment (bergeron) [inline extensions/arc/src/ui/tree/controllerTreeNode.ts:68]: Could this filter for the instance type too?  It's possible to have a pg and miaa with the same namespace and name.

Comment (bergeron) [reply extensions/arc/src/ui/tree/controllerTreeNode.ts:68]: Actually I'm not sure you can in practice.  But it'd be good to filter on.

Comment (Charles-Gagnon) [reply extensions/arc/src/ui/tree/controllerTreeNode.ts]: Yeah I think the namespace/name are at the k8 level so probably can't. Still easy enough to add so might as well.

Comment (bergeron) [reply extensions/arc/src/ui/tree/controllerTreeNode.ts]: k8s lets you have resources with the same name and namespace if the resources are different types (which is the case for sql and pg).

**Pull request merged**